### ISSUE TITLE
Support WebUI on KernelSU

### ISSUE THREAD

**Issue by ManyullynSword** [mywalkb/LSPosed_mod#65]
### Is your feature request related to a problem?/你的请求是否与某个问题相关？

You can open LSPosed Manager on the modules activity of KernelSU.

### Describe the solution you'd like/描述你想要的解决方案

I can open LSPosed Manager activity on the modules activity of KernelSU

**Comment by ManyullynSword** [comment]
This has two advantages, one is that it can eliminate the parasitic manager, and can further hide LSPosed and compact modules; the other is that it can be compatible with WSA, because WSA cannot fix shortcuts, it is very inconvenient to open LSPosed Manager from the notification and it is also a one-time, with WebUI, you can open LSPosed more conveniently, and I think WebUI will definitely support shortcuts in the future, and it will not affect the use at that time.

**Comment by mywalkb** [comment]
To eliminate the parasitic manager, we need to implement the manager in html via webui and use the cli to communicate with the lspd daemon. It's possible but it's not an immediate job.
Opening the parasitic manager or the manager installed via webui should be simple.
I'm studying to understand how to implement it, I need to understand how to use the ksu module.

**Comment by ManyullynSword** [comment]
> To eliminate the parasitic manager, we need to implement the manager in html via webui and use the cli to communicate with the lspd daemon. It's possible but it's not an immediate job. Opening the parasitic manager or the manager installed via webui should be simple. I'm studying to understand how to implement it, I need to understand how to use the ksu module.
Actually, if you think about it, it's not too feasible to eliminate the parasitic manager, and I ignored the users of LSPosed and Magisk. But this is important for WSA users. If the WebUI is supported, users will no longer need to open LSPosed Manager via batch files. In addition, the WebUI does not open the original Manager, because it is not an activity parasitic in the shell, but a separate WebUI.

**Comment by ManyullynSword** [comment]
> KernelSU is not available on all devices and with ZygiskNext become a close project (license changed and no more release on github but only on TG), Magisk will be always the best alternative to root and LSPosed_mod.
> 
> I'd like to eliminate parasitic manager and restore only installed manager, maybe with randome name, so just to give a chance to hide it even if it is useless, since the techniques to detect XPosed are now advanced.
> 
> The webui manager alone would be useful to few, however if someone wants to develop it I will accept PR without problems.
> 
> Now the opening of the parasite manager is implemented. I need to find a way to automate the compilation of html and js files with parceljs and I can release it. This will be how it works, do you think it is acceptable?
> 
>  screen-20240401-211512.mp4

I can accept it. I'm actually content as long as this feature is available - maybe it's something I'd like to look like in the end (I want it to be as close to vanilla manager as possible, including the module repository). But I'm not a developer after all, and it's great that my ideas are adopted. As for removing the parasitic manager and returning to the apk manager, I can only say that there are pros and cons. I haven't done Android development, so I don't know how to implement random package names, and random package names will also cause some trouble. There is a very powerful group in China called 360, and they have done a lot of software, including apk reinforcement. The 360 hardened apk now prevents XPosed module hooks and detects the KernelSU environment (up to the end of the method). I don't know what the consequences of removing the parasitic manager will ultimately be, but I don't think that's the best solution.

**Comment by privacyguy123** [comment]
This is very interesting to me. Could you update us if you are able to find a way to strip the parasitic manager out completely and just have it running as a KSU webui?

**Comment by mywalkb** [comment]
as has been said in this issue, the manager is also used on Magisk and Webui exists only on KernelSU, one could think of making the parasitic manager or the installed manager optional and leaving only the CLI, so the manager that will be developed on WebUI can use the CLI to manage LSPosed.
At the moment it is not in my plans to develop a manager with WebUI, but I accept PR.
I stopped using KernelSU due to a problem on my device so I went back to Magisk and now even if I could go back, the ZygiskNext license change worries me. Furthermore, it has become a closed-source project, it uses github only to release the already compiled binaries, it is out of the spirit that moves the open-source community.
KernelSU should have a built-in zygisk like Magisk.
